drivers/pds/lib/utils.go

```diff
@@ -814,6 +817,204 @@ func GetDeploymentCredentials(deploymentID string) (string, error) {
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": deploymentName},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "tpcc-run",
+							Image: pdsTpccImage,
+							Command: []string{"/bin/sh", "-C", fileToRun, dbUser, pdsPassword, dnsEndpoint, dbName,
+								timeToRun, numOfThreads, numOfCustomers, numOfWarehouses, "run"},
+							WorkingDir: "/sysbench-tpcc",
+						},
+					},
+					InitContainers: []corev1.Container{
+						{
+							Name:  "tpcc-prepare",
+							Image: pdsTpccImage,
+							Command: []string{"/bin/sh", "-C", fileToRun, dbUser, pdsPassword, dnsEndpoint, dbName,
+								timeToRun, numOfThreads, numOfCustomers, numOfWarehouses, "prepare"},
+							WorkingDir:      "/sysbench-tpcc",
+							ImagePullPolicy: corev1.PullAlways,
+						},
+					},
+					RestartPolicy: corev1.RestartPolicyAlways,
+				},
+			},
+		},
+	}
+	log.InfoD("Going to Trigger TPCC Workload for the Deployment")
+	deployment, err := k8sApps.CreateDeployment(deploymentSpec, metav1.CreateOptions{})
+	if err != nil {
+		log.Errorf("An Error Occured while creating deployment %v", err)
+		return false
+	}
+
+	timeAskedToRun, err := strconv.Atoi(timeToRun)
+	flag := false
+	// Hard sleep for 10 seconds for deployment to come up
+	time.Sleep(10 * time.Second)
+	var newPods []corev1.Pod
+	for i := 1; i <= 200; i++ {
+		newPodList, _ := GetPods(namespace)
+		newPods = append(newPods, newPodList.Items...)
+		for _, pod := range newPods {
+			if strings.Contains(pod.Name, deployment.Name) {
+				log.InfoD("Will check for status of Init Container Once......")
+				for _, c := range pod.Status.InitContainerStatuses {
+					if c.State.Terminated != nil {
+						flag = true
+					}
+				}
+			}
+		}
+		if flag {
+			log.InfoD("TPCC Schema Prepared successfully. Moving ahead to run the TPCC Workload now.....")
+			break
+		} else {
+			log.InfoD("Init Container is still running means TPCC Schema is being prepared. Will wait for further 30 Seconds.....")
+			time.Sleep(30 * time.Second)
+		}
+	}
+	if !flag {
+		log.Errorf("TPCC Schema couldn't be prepared in 100 minutes. Timing Out. Please check manually.")
+		return false
+	}
+	flag = false
+	for i := 1; i <= int((timeAskedToRun+300)/60); i++ {
+		newPodList, _ := GetPods(namespace)
+		newPods = append(newPods, newPodList.Items...)
+		for _, pod := range newPods {
+			if strings.Contains(pod.Name, deployment.Name) {
+				log.InfoD("Waiting for TPCC Workload Container to finish")
+				for _, c := range pod.Status.ContainerStatuses {
+					if int32(c.RestartCount) != 0 {
+						flag = true
+						if c.State.Terminated != nil && c.State.Terminated.ExitCode != 0 && c.State.Terminated.Reason != "Completed" {
+							log.Errorf("Something went wrong and Run Container Exited abruptly. Leaving the TPCC deployment as is - pls check manually")
+							return false
+						}
```

Comment:
you can print deployment describe oupout here

Reply:
Done